Heretic multi-patch and dejankification

misterghast opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
The focus glitch has been fixed, as the heretic magic ability check not uses the inventory RelayEvent instead of the HandSystem RelayEvent. I have also made it so that the Mansus Grasp can be toggled with the ability (turning it off does not activate it's cooldown, unlike turning it on), and that the magic contraband text of heretic items now properly displays instead of the localization string. 

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
The mansus grasp tweak was basically already in the game (you can just use it on any old non-living object to get rid of it) and in the few situations where you couldn't put it away, you really should have been able to.  The rest is clearly a restoration of intended behavior. 

## Technical details
<!-- Summary of code changes for easier review. -->
Made mansus grasp action delete the grasp when used with it active. This does not set it as handling the action, so the action does not get put on cooldown from this.

The contraband issue seemed to have been caused by a purposely null line in the resource/definition/whatever file, which was probably a holdover from that line needing to be included.

The focus glitch was fixed by just making it check the player's inventory as well as the hand, by adding the SharedInventorySystem.Relay to the subscribers for the magic focus check event. 

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
![image](https://github.com/user-attachments/assets/9f076b5f-b052-4002-9e95-878a27c21eb2)

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [ x ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [ x ] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
There is a possibility that the contraband text fix could conflict with what was done for the blood cultist contraband, but that should be easy to resolve.

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added the ability to disable the Mansus grasp by activating the ability when it's active. Disabling it's ability does not trigger the cooldown, as it does not count as actually triggering the ability like summoning the hand does.
- fix: Fixed heretic contraband legality text
- fix: Fixed heretic focus needing to be in a hand slot only to work, now a focus can be any slot except pockets, allowing the necklace and hood to function.
-->

taydeo commented on Content.Server/_Goobstation/Heretic/EntitySystems/HereticMagicItemSystem.cs:
```diff
@@ -24,11 +24,11 @@ private void OnCheckMagicItem(Entity<HereticMagicItemComponent> ent, ref HeldRel
 
-    private void OnMagicItemExamine(Entity<HereticMagicItemComponent> ent, ref ExaminedEvent args)
+    /*private void OnMagicItemExamine(Entity<HereticMagicItemComponent> ent, ref ExaminedEvent args)
     {
```
Do not comment code. Just delete it.

taydeo commented on Content.Server/_Goobstation/Heretic/Abilities/HereticAbilitySystem.cs:
```diff
@@ -145,7 +146,15 @@ private void OnMansusGrasp(Entity<HereticComponent> ent, ref EventHereticMansusG
         if (ent.Comp.MansusGraspActive)
         {
-            _popup.PopupEntity(Loc.GetString("heretic-ability-fail"), ent, ent);
+            //_popup.PopupEntity(Loc.GetString("heretic-ability-fail"), ent, ent);
```
Same here. Do not comment code, just delete it.

JoulesBerg commented:
I would also say leaving the action unhandled on cancel could cause problems. I haven't looked specifically at the mansus grasp code to know if it WOULD, but I think its a bad practice in general. Leaving events or actions unhandled I think should be avoided - some events and actions run in a continuous loop until handled, which could use up resources, etc. In some cases this could even be a memory leak (situtation where the client doesn't realize theres an unhandled event, or there is no longer a legitimate code path to mark the event as handled)

Personally I would say its better to just mark it as handled and initiate the cooldown - this would also act as a deterrent / punishment for just randomly activating the ability.

misterghast commented:
OK. I removed the commented code completely and changed how the manus grasp's cooldown works so the majority of the cooldown is now applied when the actual grasp is used (there is still a two second cooldown for enabling and disabling it, to stop spam). I did this instead of just applying the full cooldown on disable because that would encourage users to mansus grasp inanimate objects which I want to avoid.
